### ISSUE TITLE
Things Gateway on LXD container

### ISSUE THREAD

**Issue by flongo82** [mozilla-iot/gateway#977]
I've successfully installed the Things Gateway on an LXD container with Ubuntu 17.10. Just wanted to share a few issues: 

- yarn does not seem to work with python3 so I had to install python2.7 and make npm point to it: 

```
sudo apt install python2.7 python-pip
npm config set python `which python2.7`
```

- The Mozilla's provided tunneling service does not seem to work as my browser keeps trying to connect and eventually times out. The name is correctly resolved so I do not think it is a DNS problem. More likely it is something related to the fact that the gateway is running behind several levels of NAT. Is there any documentation about how the tunneling service actually work? Any idea about how I can solve the issue? 

- I correctly received a mail confirmation request but when I click on the provided link I'm redirected to the following message: 

```
An error happened while verifying your email.
```

I'm wondering if this is related to the fact that the gateway is unreachable.

**Comment by mrstegeman** [comment]
The tunneling service is documented here: https://github.com/mozilla-iot/registration_server/blob/master/doc/flow.md

After initial domain registration and stuff, it's essentially just a [PageKite](https://pagekite.net/) tunnel, so your problem is probably with PageKite. PageKite errors should be logged to `~/.mozilla-iot/logs/run-app.log`, so you may want to check that and/or pastebin it for us to look at.

The email verification issue is something we've seen before with our EU server, but I've been unable to track it down. Would you mind sharing the domain you're trying to set up so I can check out our logs?

**Comment by flongo82** [comment]
Hi and thanks for your reply! 

There is no `~/.mozilla-iot/logs/run-app.log` in my system. Actually, there is only a `~/.mozilla-iot/log` folder but it is empty. Is it because I'm not using the flash image but I'm compiling the gateway from scratch? 

However, I realized that the tunneling correctly work when I'm home while it does not when I'm at work. This enforces my feeling that it is mainly a NAT/Firewall problem. 

Regarding the email verification problem, the domain is smartme.mozilla-iot.org. 

An additional problem (do not know if related to the fact that I'm running in an LXD container): I added the thing-url-adapter and connected a Web Thing running inside another LXD container. The gateway is continuously printing the following message in the console: 

```
2018-05-04 10:25:36.843 Plugin: thing-url-adapter died, code = null restarting...
2018-05-04 10:25:37.023 thing-url: Opening database: /home/ubuntu/.mozilla-iot/config/db.sqlite3
2018-05-04 10:25:37.035 thing-url: Loading add-on for thing-url-adapter from /home/ubuntu/.mozilla-iot/addons/thing-url-adapter
2018-05-04 10:25:37.226 thing-url: node: ../deps/uv/src/unix/poll.c:120: uv_poll_start: Assertion `!(((handle)->flags & (UV_CLOSING | UV_CLOSED)) != 0)' failed.

```
